blocks/utils.py

```diff
@@ -429,3 +429,47 @@ def ipdb_breakpoint(x):
     """
     import ipdb
     ipdb.set_trace()
+
+
+class LambdaIterator(six.Iterator):
+    """An iterator that calls a function to fetch the next element.
+
+    The reason for having this is that generators are not serializable
+    in Python (even when using third-party libraries).
+
+    Parameters
+    ----------
+    next_function : callable
+        A function to call every time the next element is requested.
+
+    """
+    def __init__(self, next_function):
+        self.next_function = next_function
+
```

Comment:
You'll need to add
```python
def __iter__(self):
    return self
```
(`six.Iterator` just maps `next` to `__next__`, nothing else)